packages/mc-scripts/src/config/webpack-dev-server.config.js

```diff
@@ -45,14 +45,12 @@ module.exports = ({ port, publicPath }) => ({
   hot: true,
   https: false,
   port,
-  // Enable HTTPS if the HTTPS environment variable is set to 'true'
-  // `proxy` is run between `before` and `after` `webpack-dev-server` hooks
   setupMiddlewares(middlewares, devServer) {
     if (!devServer) {
       throw new Error('webpack-dev-server is not defined');
     }
 
-    middlewares.push(createMcDevAuthenticationMiddleware(applicationConfig));
+    devServer.app.use(createMcDevAuthenticationMiddleware(applicationConfig));
```

Comment:
This is the fix. Apparently using the `middlewares` list didn't really work as expected.

Reply:
Woah, I read the documentation to verify but it also does this. When debugging the production build on the middlewares, the only thing I noticed it that it wasn't called.